ocaml/runtime/obj.c

```diff
@@ -161,11 +161,22 @@ CAMLprim value caml_obj_with_tag(value new_tag_v, value arg)
     res = caml_alloc_small(sz, tg);
     for (i = 0; i < sz; i++) Field(res, i) = Field(arg, i);
   } else {
+    mlsize_t scannable_sz = Scannable_wosize_val(arg);
+
     res = caml_alloc_shr(sz, tg);
     /* It is safe to use [caml_initialize] even if [tag == Closure_tag]
        and some of the "values" being copied are actually code pointers.
        That's because the new "value" does not point to the minor heap. */
-    for (i = 0; i < sz; i++) caml_initialize(&Field(res, i), Field(arg, i));
+    for (i = 0; i < scannable_sz; i++) {
+      caml_initialize(&Field(res, i), Field(arg, i));
+    }
+
+    if (scannable_sz < sz) {
+      memcpy(Bp_val(res) + scannable_sz,
+             Bp_val(arg) + scannable_sz,
```

Comment:
I think those `Bp_val` should be `Op_val`, since `scannable_sz` is in words, not bytes.

Reply:
Indeed.